### ISSUE TITLE
prettier should be a dev dep

### ISSUE THREAD

**Issue by michaelfarrell76** [storybookjs/storybook#14714]
**Describe the bug**
In the following: prettier is a regular dependency when it should be a devDependency
- "@storybook/addon-docs@npm:6.2.9":
- "@storybook/codemod@npm:6.2.9": 
- "@storybook/source-loader@npm:6.2.9":

**Comment by shilman** [comment]
Why should prettier be a dev dep? Storybook directly depends on it to function.

**Comment by JanJakes** [comment]
Having prettier as a dependency indeed can cause problems when your own projects wants to use prettier as well (which I guess is often the case). We use prettier `2.3.1` in our monorepo in the workspace root but we've been seeing random small differences in formatting when used by different devs or in the CI pipeline.

After some investigation we realized this was caused by the Storybook's prettier sometimes being loaded instead of "our" prettier (which can indeed happen with NPM and Yarn v1). We had to use the `resolutions` field to enforce only one version of prettier but now we're getting `warning Resolution field "prettier@2.3.1" is incompatible with requested version "prettier@~2.2.1"`.

I think there are two possible ways to solve this:
1) The very quick & easy one is to use `^2.2.1` instead of `~2.2.1` in Storybook packages. I think this should cause no problems as prettier almost never changes its config and API.
2) Make prettier a peer dependency (but then the question is how to handle a missing install plus point 1) still stands here, especially with NPM 7 that installs peer deps automatically).

---

**TL;DR:** Could Storybook require prettier `^2.2.1` instead of `~2.2.1`?

**Comment by shilman** [comment]
@ndelangen do you know why we're using "~2.2.1" instead of "^2.2.1"? you changed it last.

**Comment by ndelangen** [comment]
I do not see a reason for this at all. Let's upgrade!
@JanJakes would you be able to open a PR upgrading us to the more lenient version-range?

**Comment by michaelfarrell76** [comment]
ah i didnt realize storybook depends on prettier to function. i thought prettier was just a dev dependency used to format the code before it's bundled, not something that is required to actually run the storybook dependencies
